packages/widgets/src/menubar.ts

```diff
@@ -699,7 +699,8 @@ export class MenuBar extends Widget {
     }
 
     // Get position for the new menu >before< updating active index.
-    const position = this._positionForMenu(index);
+    const position =
+      index >= 0 && this._childMenu ? this._positionForMenu(index) : null;
```

Comment:
I am not sure if `index >= 0` ever gets `false`: the case of `-1` is handled in line 697 and `ArrayExt.findFirstIndex` should always return a non-negative number or `-1`.

But then I am also not sure why `itemNode` would ever be undefined (what would mutate `this.contentNode.children` between line 685 and 702?). Conditioning on `this._childMenu` seems fine but I don't have the full picture right now.

Reply:
Oh, I see  line 697 only short-circuits if index is -1 and child_menu is already open. Ok so this makes sense.